libafl/src/stages/sync.rs

```diff
@@ -164,28 +162,23 @@ where
 impl<CB, E, EM, Z> SyncFromDiskStage<CB, E, EM, Z> {
     /// Creates a new [`SyncFromDiskStage`]
     #[must_use]
-    pub fn new(sync_dir: PathBuf, load_callback: CB, name: &str) -> Self {
+    pub fn new(sync_dirs: Vec<PathBuf>, load_callback: CB, interval: Duration, name: &str) -> Self {
         Self {
             name: Cow::Owned(SYNC_FROM_DISK_STAGE_NAME.to_owned() + ":" + name),
             phantom: PhantomData,
-            sync_dir,
+            sync_dirs,
+            interval,
             load_callback,
         }
     }
-
+    #[allow(clippy::only_used_in_recursion)]
```

Comment:
False positive? oO

Reply:
``` rust

error: parameter is only used in recursion
   --> libafl/src/stages/sync.rs:176:10
    |
176 |         &self,
    |          ^^^^
    |
note: parameter used here
   --> libafl/src/stages/sync.rs:203:40
    |
203 |                 let dir_left_to_sync = self.load_from_directory(&entry.path(), last)?;
    |                                        ^^^^
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#only_used_in_recursion
note: the lint level is defined here
   --> libafl/src/lib.rs:18:9
    |
18  | #![deny(clippy::all)]
    |         ^^^^^^^^^^^
    = note: `#[deny(clippy::only_used_in_recursion)]` implied by `#[deny(clippy::all)]`

```

Reply:
Ok technically `load_from_directory` could be a normal `fn` instead of being a method

Reply:
So the clippy lint is correct

Reply:
Take a look at #2402 what you think